### PR TITLE
Add examples with Kotlin-DSL

### PR DESCRIPTION
**Pull request by StefMa**
Hi guys 👋 

I've just added some examples for the kotlin-dsl.

Some things to mention:
* I don't know if it's recommended to use `closureOf<SubTaskName>` for the task topic. (See also https://github.com/gradle/kotlin-dsl/issues/880) cc @eskatos 
* I can't really update the test (like you have done here https://github.com/gradle-guides/creating-build-scans/pull/12/files#diff-6c874baa6043fd24764005086d884bde) cause it fail anyway (the `import` statement  is not at top of the file). 

Beside of that. Great work @eriwen with the docs. It is really easy to migrate 👌

### CODE REVIEW

File: contents/index.adoc [hunk 1]
```diff
@@ -65,12 +65,17 @@ You now have a plugin, but a plugin alone doesn't do anything. You need to apply
 
 == Use the plugin in the main project
 
-Create a `build.gradle` file in the root of the project with the following contents:
+Create a `build.gradle` (or a `guild.gradle.kts` respectively) file in the root of the project with the following contents:
```
Comment (eskatos) [inline contents/index.adoc]: :x: rephrase to `Create a build script file in the ...`
The two snippets are titled with the file name to use.

File: contents/index.adoc [hunk 2]
```diff
@@ -122,17 +132,29 @@ WARNING: You must put the properties file in the directory `META-INF/gradle-plug
...
+.build.gradle.kts
+----
+include::{exampledir}/build.gradle.kts[tags=use-plugin-class-name]
 ----
 
 with one that uses the plugin ID:
 
-[source,groovy]
+[source.multi-language-sample,groovy]
+.build.gradle
```
Comment (eskatos) [inline contents/index.adoc]: :x: move the snippet title on top, e.g:
```
.build.gradle
[source.multi-language-sample,groovy]
----
include::...
----
This applies to all snippets in this file.

File: samples/code/configure-hello.gradle [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // tag::use-plugin-id[]
-apply plugin : 'org.example.greeting'
+apply plugin: 'org.example.greeting'
```
Comment (eskatos) [inline samples/code/configure-hello.gradle]: :x: please prefer using the `plugins {}` block in both Groovy and Kotlin samples when applying by plugin ID

File: samples/code/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+// tag::use-plugin-class-name-with-info[]
+apply {
```
Comment (eskatos) [inline samples/code/build.gradle.kts]: :x: should be `apply<org.example.greeting.GreetingPlugin>()` instead

File: samples/code/configure-hello.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// tag::use-plugin-id[]
+apply {
+  plugin("org.example.greeting")
+}
+// end::use-plugin-id[]
+
+// tag::configure-hello[]
+import org.example.greeting.Greeting
```
Comment (eskatos) [inline samples/code/configure-hello.gradle.kts]: :x: move the import to the top of the file, you can tag/end multiple times with the same tag:

```
// tag::configure-hello[]
import org.example.greeting.Greeting
// end::configure-hello[]

// tag::use-plugin-id[]
plugins {
    id("org.example.greeting")
}
// end::use-plugin-id[]

// tag::configure-hello[]
// the rest goes here
// end::configure-hello[]
```

File: samples/code/configure-hello.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+tasks.getByName("hello", closureOf<Greeting> { // <1>
```
Comment (eskatos) [inline samples/code/configure-hello.gradle.kts]: :x: should be:
```
tasks.getByName<Greeting>("hello") {
    // ...
}
```

File: contents/index.adoc (hunk 2 above)
Comment (StefMa) [reply contents/index.adoc]: I've done that. But I've more or less followed your PR here: https://github.com/gradle-guides/creating-build-scans/pull/12

Both have the same output/outcome, why is which preferred? 🤔

File: samples/code/configure-hello.gradle.kts (hunk 5 above)
Comment (StefMa) [reply samples/code/configure-hello.gradle.kts]: Wow. That is really cool 👍

File: samples/code/configure-hello.gradle.kts (hunk 6 above)
Comment (StefMa) [reply samples/code/configure-hello.gradle.kts]: This isn't working 🤔 
Seems that this extension is not available...
bildschirmfoto 2018-05-25 um 14 56 47

Run this with Gradle 4.7 stable.

Comment (StefMa) [reply samples/code/configure-hello.gradle.kts]: Just tested it with Gradle 4.8-RC2. 
Works with that.

What should I do?
Update the wrapper here, revert the changes or ...?! 🤔

Comment (eskatos) [reply samples/code/configure-hello.gradle.kts]: Right. Yes please update the wrapper to 4.8-rc-2 in this PR, we'll update it to 4.8 once GA is out and merge with it.

File: src/test/groovy/CompilePluginSpec.groovy [hunk 7]
```diff
@@ -15,17 +15,40 @@ class CompilePluginSpec extends AbstractSamplesFunctionalTest {
         initialize()
     }
 
-    void 'Validate example code runs'() {
+    void 'Validate example code runs with #lang'() {
```
Comment (eskatos) [inline src/test/groovy/CompilePluginSpec.groovy]: :x: missing `@Unroll` annotation

File: src/test/groovy/CompilePluginSpec.groovy [hunk 8]
```diff
@@ -15,17 +15,40 @@ class CompilePluginSpec extends AbstractSamplesFunctionalTest {
...
         output.contains(':hello')
         output.contains('Hello, World')
+
+        where:
+        lang     | buildScriptFilename
+        'Groovy' | 'build.gradle'
+        'Kotlin' | 'build.gradle.kts'
+    }
+
+    void 'Validate configure task runs with #lang'() {
```
Comment (eskatos) [inline src/test/groovy/CompilePluginSpec.groovy]: :x: missing `@Unroll` annotation

File: contents/index.adoc (hunk 2 above)
Comment (StefMa) [reply contents/index.adoc]: I've dropped that commit by rebasing the branch 👍

**Comment by StefMa** [comment]
Update to gradle 4.8 🎉 
And rebased duo of conflicts 🤘

**Comment by eriwen** [comment]
You should link to https://docs.gradle.org/current/userguide/organizing_gradle_projects.html. I have submitted a change to redirect to organizing_gradle_projects now.

**Comment by StefMa** [comment]
@eriwen @eskatos 
Anything which hides you to merge this? 😇

**Pull request merged**